### PR TITLE
SUBMARINE-314. Creating a k8s environment locally using kind

### PR DESCRIPTION
**Pull request by liuxunorg**
### What is this PR for?
Because the submarine on cloud is based on the k8s operating environment, in order to be able to develop and test locally, we use kind to deploy a complete k8s environment locally.
1. install k8s cluster
2. install docker registry

### What type of PR is it?
Feature

### Todos
* [ ] - Task

### What is the Jira issue?
* https://issues.apache.org/jira/browse/SUBMARINE-314

### How should this be tested?
* [CI Pass](https://travis-ci.org/liuxunorg/submarine/builds/622469321)

### Screenshots (if appropriate)
image

```
kubectl get pods -A
NAMESPACE     NAME                       READY   STATUS    RESTARTS   AGE
kube-system   coredns-576cbf47c7-mk5bb   0/1     Pending   0          27s
kube-system   coredns-576cbf47c7-zbhgz   0/1     Pending   0          27s
kube-system   ip-masq-agent-fdzr5        1/1     Running   0          27s
kube-system   ip-masq-agent-w2pgc        1/1     Running   0          25s
kube-system   kindnet-hbclv              1/1     Running   0          25s
kube-system   kindnet-vlx47              1/1     Running   0          27s
kube-system   kube-proxy-84c9z           1/1     Running   0          25s
kube-system   kube-proxy-mvvbz           1/1     Running   0          27s
```

```
docker tag busybox localhost:5000/busybox
docker push localhost:5000/busybox
The push refers to repository [localhost:5000/busybox]
1da8e4c8d307: Pushed 
latest: digest: sha256:679b1c1058c1f2dc59a3ee70eed986a88811c0205c8ceea57cec5f22d2c3fbb1 size: 527
```

### Questions:
* Does the licenses files need update? Yes/No
* Is there breaking changes for older versions? Yes/No
* Does this needs documentation? Yes/No

### CODE REVIEW

File: submarine-cloud/hack/kind-cluster-build.sh [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+cd $ROOT
+
+source $ROOT/hack/lib.sh
+
+hack::ensure_kubectl
+hack::ensure_kind
+
+usage() {
+    cat <<EOF
+This script use kind to create Kubernetes cluster,about kind please refer: https://kind.sigs.k8s.io/
```
Comment (yuanzac) [inline submarine-cloud/hack/kind-cluster-build.sh]: . Please refer to http... for kind details

File: submarine-cloud/hack/kind-cluster-build.sh [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+Before run this script,please ensure that:
+* have installed docker
```
Comment (yuanzac) [inline submarine-cloud/hack/kind-cluster-build.sh]: docker installed

File: submarine-cloud/hack/kind-cluster-build.sh [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+* have installed kind and kind's version == v0.6.0
```
Comment (yuanzac) [inline submarine-cloud/hack/kind-cluster-build.sh]: kind installed

File: submarine-cloud/hack/kind-cluster-build.sh [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+
+source $ROOT/hack/lib.sh
+
+hack::ensure_kubectl
+hack::ensure_kind
+
+usage() {
+    cat <<EOF
+This script use kind to create Kubernetes cluster,about kind please refer: https://kind.sigs.k8s.io/
+Before run this script,please ensure that:
```
Comment (yuanzac) [inline submarine-cloud/hack/kind-cluster-build.sh]: Add a space before "please"

**Comment by liuxunorg** [comment]
Will merge if no more comments.

**Pull request closed without merging**